common/models/report.js

```diff
@@ -1,12 +1,84 @@
-var countries = require('country-data').countries;
+var axios = require('axios')
+var qs = require('qs')
+
+var countries = require('country-data').countries
+
+var apiClient = axios.create({
+  // XXX Uncomment this when we deploy it
+  // baseURL: 'https://api.ooni.io/api/',
+  // XXX We use this base URL for testing
+  baseURL: 'http://localhost:3001/api/',
+  timeout: 90000, // Maybe set this lower once performance is boosted
+})
+
 module.exports = function(Report) {
 
+  Report.findMeasurements = function(probe_cc, input, order, page_number,
+      page_size, since, until, test_name, callback) {
+    var apiQuery = {}
+    if (probe_cc) {
+      apiQuery.probe_cc = probe_cc
+    }
+    if (input) {
+      apiQuery.input = input
+    }
+    /*
+     * XXX currently disabled due to gateway timeouts
+     * cc @darkk
+    if (order) {
+      apiQuery.order_by = order.split(' ')[0]
+      apiQuery.order = order.split(' ')[1]
+    }
+    */
+    if (page_number && page_size) {
+      apiQuery.offset = page_number * page_size
+      apiQuery.limit = page_size
+    }
```

Comment:
> `cc @darkk`
He-he. `offset` will also lead to timeout when huge `offset`s are used :-)
That's what we've already discussed today.